livedata/src/test/java/io/getstream/chat/android/livedata/controller/QueryChannelsControllerImplTest.kt

```diff
@@ -0,0 +1,91 @@
+package io.getstream.chat.android.livedata.controller
+
+import androidx.arch.core.executor.testing.InstantExecutorExtension
+import com.nhaarman.mockitokotlin2.*
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.livedata.ChatDomainImpl
+import io.getstream.chat.android.livedata.utils.getOrAwaitValue
+import kotlinx.coroutines.Job
+import org.amshove.kluent.shouldBeEqualTo
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+
+@ExtendWith(InstantExecutorExtension::class)
+internal class QueryChannelsControllerImplTest {
+    @Test
+    fun `when add channel if filter matches should update LiveData from channel to channel controller`() {
+        val channelController = mock<ChannelControllerImpl>()
+        val sut = Fixture()
+            .givenNewChannelController(channelController)
+            .setupChatControllersInstantiation()
+            .get()
+        val newChannel = Channel(cid = "CID", id = "ID")
+
+        sut.addChannelIfFilterMatches(newChannel)
+
+        verify(channelController).updateLiveDataFromChannel(eq(newChannel))
+    }
+
+    @Test
+    fun `when add channel if filter matches should post value to liveData with the same channel ID`() {
+        val sut = Fixture()
+            .givenNewChannelController(mock())
+            .setupChatControllersInstantiation()
+            .get()
+        val newChannel = Channel(cid = "ChannelType:ChannelID", id = "ChannelID")
+
+        sut.addChannelIfFilterMatches(newChannel)
+
+        val result = sut.channels.getOrAwaitValue()
+        result.size shouldBeEqualTo 1
+        result.first().cid shouldBeEqualTo "ChannelType:ChannelID"
+    }
+
+    @Test
+    fun `when add channel twice if filter matches should post value to liveData only one value`() {
+        val sut = Fixture()
+            .givenNewChannelController(mock())
+            .setupChatControllersInstantiation()
+            .get()
+        val newChannel = Channel(cid = "ChannelType:ChannelID", id = "ChannelID")
+
+        sut.addChannelIfFilterMatches(newChannel)
+        sut.addChannelIfFilterMatches(newChannel)
+
+        val result = sut.channels.getOrAwaitValue()
+        result.size shouldBeEqualTo 1
+        result.first().cid shouldBeEqualTo "ChannelType:ChannelID"
+    }
+}
+
+class Fixture {
```

Comment:
I think this could be `private` within this file, if it's only meant for use in this one test

Reply:
You're right